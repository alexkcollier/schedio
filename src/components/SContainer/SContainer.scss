.container {
  margin: 0 auto;
  padding: spacing();
  width: 100%;

  &:not(.container--fullwidth) {
    @each $breakpoint in map-keys($media-breakpoints) {
      @include after-breakpoint($breakpoint) {
        max-width: map-get($container-breakpoints, $breakpoint);
      }
    }
  }
}
