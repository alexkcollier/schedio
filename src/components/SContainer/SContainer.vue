<template>
  <Component
    :is="tag"
    :class="{ 'container--fullwidth': isFullwidth }"
    class="container"
  >
    <slot />
  </Component>
</template>

<script>
import defaults from '@/utils/defaults'

export default {
  name: 'SContainer',

  props: {
    tag: {
      type: String,
      default: () => defaults.containerTag,
    },
    isFullwidth: {
      type: Boolean,
      default: false,
    },
  },
}
</script>

<style lang="scss">
.container {
  margin: 0 auto;
  padding: spacing();
  width: 100%;

  &:not(.container--fullwidth) {
    @each $breakpoint in map-keys($media-breakpoints) {
      @include after-breakpoint($breakpoint) {
        max-width: map-get($container-breakpoints, $breakpoint);
      }
    }
  }
}
</style>
