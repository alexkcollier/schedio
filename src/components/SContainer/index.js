import SContainer from './SContainer.vue'

export { SContainer }

export default { install: Vue => Vue.component(SContainer.name, SContainer) }
