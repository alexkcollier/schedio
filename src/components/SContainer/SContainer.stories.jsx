import { SContainer } from '@/components/SContainer'
import { SContainerColumn } from '@/components/SContainerColumn'
import { SContainerRow } from '@/components/SContainerRow'
import { SHeading } from '@/components/SHeading'
import { boolean } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/vue'
import PropList from '@/docs/PropList'

storiesOf('Components/Layout.SContainer', module)
  .addParameters({ jest: 'SContainer' })
  .add('SContainer', function SContainerDocs() {
    return {
      props: {
        props: {
          default: () => ({
            isFullwidth: boolean('is-fullwidth', false)
          })
        }
      },
      render(h) {
        const { props } = this.$props
        return (
          <div>
            <SHeading level="1">Container</SHeading>
            <p>
              The container defaults to being narrower than the window, but it can be set to
              fullwidth. It has been styled for visibility, and is normally invisible.
            </p>
            <p>
              It has padding that is hidden when used with a <code>SContainerRow</code>.
            </p>

            <SHeading level="2">Example</SHeading>

            <SContainer {...{ props }} style="border: 1px solid black;">
              <SContainerRow>
                <SContainerColumn>Container content</SContainerColumn>
              </SContainerRow>
            </SContainer>

            <PropList component={SContainer} />
          </div>
        )
      }
    }
  })
