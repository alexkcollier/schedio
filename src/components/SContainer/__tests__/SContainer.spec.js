import { shallowMount } from '@vue/test-utils'
import SContainer from '@/components/SContainer/SContainer.vue'

describe('SContainer.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(SContainer, {
      slots: {
        default: '<div>Container Content</div>',
      },
    })
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders slot content', () => {
    expect(wrapper.vm.$slots.default).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can be fullwidth', () => {
    wrapper.setProps({ isFullwidth: true })
    expect(wrapper.vm.isFullwidth).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
