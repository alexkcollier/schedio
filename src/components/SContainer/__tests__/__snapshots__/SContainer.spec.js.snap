// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SContainer.vue can be any html element 1`] = `
<section class="container">
  <div>Container Content</div>
</section>
`;

exports[`SContainer.vue can be fullwidth 1`] = `
<div class="container container--fullwidth">
  <div>Container Content</div>
</div>
`;

exports[`SContainer.vue renders correctly 1`] = `
<div class="container">
  <div>Container Content</div>
</div>
`;

exports[`SContainer.vue renders slot content 1`] = `
<div class="container">
  <div>Container Content</div>
</div>
`;
